src/vivarium_cluster_tools/psimulate/distributed_worker.py

```diff
@@ -155,26 +157,58 @@ def worker(parameters: Mapping):
         logger.info('Simulation configuration:')
         logger.info(str(sim.configuration))
 
-        start = time()
+        start_time = pd.Timestamp(**sim.configuration.time.start.to_dict())
```

Comment:
This all looks like it's doing what it is supposed to, but the performance tracking is heavily interleaved with the computation which really reduces readability.  The first bit here is to make things work, of course, but we should remember to come back here an find a reasonable abstraction for doing this tracking in a way that doesn't obfuscate the computation.  

Three ideas (though there are other ways to do this as well):

- Subclass the SimulationContext and have it do the performance tracking as part of its normal operations
- Subclass the logger to to performance tracking
- Make a PerformanceObserver class that we stick in the sim here and have it track this stuff.